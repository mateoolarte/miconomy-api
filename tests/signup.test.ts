import { createTestContext } from './__helpers';

const ctx = createTestContext();

const signupMutation = `
  mutation signup($email: String!, $password: String!) {
    signup(email: $email, password: $password) {
      message
      status
      user {
        email
        id
      }
    }
  }
`;

describe('signup', () => {
  it('should signup a new user', async () => {
    const signupUser = await ctx.client.request(signupMutation, {
      email: 'test1@miconomy.co',
      password: '12345678',
    });

    expect(signupUser).toMatchSnapshot();

    const users = await ctx.db.user.findMany({
      select: {
        id: true,
        email: true,
      },
    });

    expect(users).toMatchSnapshot(`
      Array [
        Object {
          "email": "test1@miconomy.co",
          "id": 1,
        },
      ]
    `);
  });
});
