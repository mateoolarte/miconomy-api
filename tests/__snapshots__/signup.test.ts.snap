// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup should signup a new user 1`] = `
Object {
  "signup": Object {
    "message": "Se ha creado el usuario correctamente",
    "status": 202,
    "user": Object {
      "email": "test1@miconomy.co",
      "id": 1,
    },
  },
}
`;

exports[`signup should signup a new user: 
      Array [
        Object {
          "email": "test1@miconomy.co",
          "id": 1,
        },
      ]
     1`] = `
Array [
  Object {
    "email": "test1@miconomy.co",
    "id": 1,
  },
]
`;
